modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/imageio/MaskOverviewProvider.java

```diff
@@ -127,45 +128,48 @@ public MaskOverviewProvider(DatasetLayout layout, URL inputFile, SpiHelper spiPr
         this.layout = layout;
 
         // Handling Overviews
-        ovrURL = new URL(inputFile.toString() + OVR_EXTENSION);
         hasDatasetLayout = layout != null;
-        if (hasDatasetLayout && layout.getExternalOverviews() != null) {
-            ovrURL = URLs.fileToUrl(layout.getExternalOverviews());
-        }
-        // Creating overview file URL
-        overviewStreamSpi =
-                suggestedStreamSPI == null ? getInputStreamSPIFromURL(ovrURL) : suggestedStreamSPI;
-        ImageInputStream ovrStream = null;
         boolean hasExternalOverviews = false;
-        try {
-            ovrStream =
-                    overviewStreamSpi.createInputStreamInstance(
-                            ovrURL, ImageIO.getUseCache(), ImageIO.getCacheDirectory());
-            if (ovrStream == null) {
-                // No Overview file so we fall back to the original file spis
-                overviewStreamSpi = streamSpi;
-                overviewReaderSpi = readerSpi;
-            } else {
-                overviewReaderSpi = getReaderSpiFromStream(null, ovrStream);
-                hasExternalOverviews = true;
+        if (!ImageIOUtilities.isSkipExternalFilesLookup()) {
```

Comment:
Note that this external IF resulted into a different indentation so you will notice several modified lines although, beside this check, the only difference is the indentation.